src/ray/util/array.h

```diff
@@ -0,0 +1,48 @@
+// Copyright 2024 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#pragma once
+
+#include <array>
+#include <cstddef>
+
+namespace ray {
+
+// Warning: O(n^2) complexity. Only use in constexpr context.
+template <typename T, size_t N>
+constexpr bool array_is_unique(const std::array<T, N> &arr) {
+  for (size_t i = 0; i < N; ++i) {
```

Comment:
Why not use `std::all_of`?
It's `constexpr` since C++ 20: https://en.cppreference.com/w/cpp/algorithm/all_any_none_of

Reply:
because we are in cpp17, sadly